cmake/UsePythonTest.cmake

```diff
@@ -20,96 +19,59 @@
+  get_source_file_property(QGIS_PYTEST_PYENV ${FILENAME} PYTHONPATH)
+
   #Avoid "NOTFOUND" string when setting LD_LIBRARY_PATH later
-  if(EXISTS "${pyenv}")
-    set(pyenv "${pyenv}:")
+  if(EXISTS "${QGIS_PYTEST_PYENV}")
+    set(QGIS_PYTEST_PYENV "${QGIS_PYTEST_PYENV}:")
   else()
-    set(pyenv "")
+    set(QGIS_PYTEST_PYENV "")
   endif()
-  IF(WIN32)
-    STRING(REGEX REPLACE ":" " " wo_semicolon "${ARGN}")
-    IF(USING_NINJA OR USING_NMAKE)
-      FILE(WRITE ${CMAKE_CURRENT_BINARY_DIR}/${TESTNAME}.cmake "
-SET(ENV{QGIS_PREFIX_PATH} \"${QGIS_OUTPUT_DIRECTORY}/bin\")
-SET(ENV{PATH} \"${QGIS_OUTPUT_DIRECTORY}/bin;\$ENV{PATH}\")
-SET(ENV{PYTHONPATH} \"${PYTHON_OUTPUT_DIRECTORY};${PYTHON_OUTPUT_DIRECTORY}/plugins;${CMAKE_SOURCE_DIR}/tests/src/python;\$ENV{PYTHONPATH}\")
-MESSAGE(\"PATH:\$ENV{PATH}\")
-")
-    ELSE(USING_NINJA OR USING_NMAKE)
-      FILE(WRITE ${CMAKE_CURRENT_BINARY_DIR}/${TESTNAME}.cmake "
-SET(ENV{QGIS_PREFIX_PATH} \"${QGIS_OUTPUT_DIRECTORY}/bin/\${CMAKE_BUILD_TYPE}\")
-SET(ENV{PATH} \"${QGIS_OUTPUT_DIRECTORY}/bin/\${CMAKE_BUILD_TYPE};\$ENV{PATH}\")
-SET(ENV{PYTHONPATH} \"${PYTHON_OUTPUT_DIRECTORY};${PYTHON_OUTPUT_DIRECTORY}/plugins;${CMAKE_SOURCE_DIR}/tests/src/python;\$ENV{PYTHONPATH}\")
-MESSAGE(\"PATH:\$ENV{PATH}\")
-")
-    ENDIF(USING_NINJA OR USING_NMAKE)
-  ELSE(WIN32)
-    STRING(REGEX REPLACE ";" " " wo_semicolon "${ARGN}")
-    FILE(WRITE ${CMAKE_CURRENT_BINARY_DIR}/${TESTNAME}.cmake "
-SET(ENV{QGIS_PREFIX_PATH} \"${QGIS_OUTPUT_DIRECTORY}\")
-SET(ENV{LD_LIBRARY_PATH} \"${pyenv}${QGIS_OUTPUT_DIRECTORY}/lib:\$ENV{LD_LIBRARY_PATH}\")
-SET(ENV{PYTHONPATH} \"${PYTHON_OUTPUT_DIRECTORY}:${PYTHON_OUTPUT_DIRECTORY}/plugins:${CMAKE_SOURCE_DIR}/tests/src/python:\$ENV{PYTHONPATH}\")
-MESSAGE(\"export LD_LIBRARY_PATH=\$ENV{LD_LIBRARY_PATH}\")
-")
-  ENDIF(WIN32)
 
-  SET(TEST_TIMEOUT 0)
+  set(_QGIS_PYTEST_PYTHONPATHS "${PYTHON_OUTPUT_DIRECTORY}" "${PYTHON_OUTPUT_DIRECTORY}/plugins" "${CMAKE_SOURCE_DIR}/tests/src/python")
 
-  FOREACH(_in ${ARGN})
-    STRING(REGEX MATCH "^([^=]+)=(.*)$" _out ${_in})
+  if(WIN32)
+    set(QGIS_PYTEST_SEPARATOR ";")
+    set(PATH_VAR_NAME "PATH")
+    if(USING_NINJA OR USING_NMAKE)
+      set(QGIS_PYTEST_PREFIX_PATH "${QGIS_OUTPUT_DIRECTORY}/bin")
+      set(QGIS_PYTEST_LIBRARY_PATH "${QGIS_OUTPUT_DIRECTORY}/bin;")
+      set(QGIS_PYTEST_PYTHONPATH "${_QGIS_PYTEST_PYTHONPATHS};")
+    else()
+      set(QGIS_PYTEST_PREFIX_PATH "${QGIS_OUTPUT_DIRECTORY}/bin/${CMAKE_BUILD_TYPE}")
+      set(QGIS_PYTEST_LIBRARY_PATH "${QGIS_OUTPUT_DIRECTORY}/bin/${CMAKE_BUILD_TYPE};")
+      set(QGIS_PYTEST_PYTHONPATH "${_QGIS_PYTEST_PYTHONPATHS};")
+    endif()
+  else()
+    set(QGIS_PYTEST_SEPARATOR ":")
+    set(QGIS_PYTEST_PATH_VAR_NAME "LD_LIBRARY_PATH")
+    set(QGIS_PYTEST_PREFIX_PATH "${QGIS_OUTPUT_DIRECTORY}")
+    set(QGIS_PYTEST_LIBRARY_PATH "${QGIS_PYTEST_PYENV}${QGIS_OUTPUT_DIRECTORY}/lib")
+    string(REPLACE ";" ":" QGIS_PYTEST_PYTHONPATH "${_QGIS_PYTEST_PYTHONPATHS};")
```

Comment:
Could this be managed as cmake list and then sent through `cmake_path(CONVERT <input> TO_NATIVE_PATH_LIST <out-var>)` (on all platforms)?